src/backend/distributed/utils/citus_clauses.c

```diff
@@ -39,6 +41,95 @@ static Node * PartiallyEvaluateExpressionMutator(Node *expression,
 static Expr * citus_evaluate_expr(Expr *expr, Oid result_type, int32 result_typmod,
 								  Oid result_collation, PlanState *planState);
 
+static bool contain_volatile_functions_checker_citus(Oid func_id, void *context);
+static bool contain_mutable_functions_checker_citus(Oid func_id, void *context);
+
+
+static bool
+contain_volatile_functions_checker_citus(Oid func_id, void *context)
+{
+	if (func_id == CitusReadIntermediateResultFuncId())
+	{
+		return false;
+	}
+
+	return (func_volatile(func_id) == PROVOLATILE_VOLATILE);
+}
+
+
+bool
+CitusContainsVolatileFunction(Node *node)
```

Comment:
`CitusIsVolatileFunction` or `CitusIsVolatileExpression` is probably more appropriate now that the function doesn't do any walking itself.